x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -407,21 +415,37 @@ export class UninstallTokenService implements UninstallTokenServiceInterface {
 
   private async getPolicyIdsBatch(
     batchSize: number = SO_SEARCH_LIMIT,
-    page: number = 1
+    page: number = 1,
+    managedPoliciesOnly?: boolean
   ): Promise<string[]> {
-    return (
-      await agentPolicyService.list(this.soClient, { page, perPage: batchSize, fields: ['id'] })
-    ).items.map((policy) => policy.id);
+    if (managedPoliciesOnly) {
+      return (
+        await agentPolicyService.list(this.soClient, {
+          page,
+          perPage: batchSize,
+          fields: ['id', 'is_managed'],
+        })
+      ).items.reduce((acc, policy) => {
+        if (policy.is_managed) {
+          acc.push(policy.id);
+        }
+        return acc;
+      }, [] as string[]);
+    } else {
+      return (
+        await agentPolicyService.list(this.soClient, { page, perPage: batchSize, fields: ['id'] })
+      ).items.map((policy) => policy.id);
+    }
   }
 
-  private async getAllPolicyIds(): Promise<string[]> {
+  private async getAllPolicyIds(managedPoliciesOnly?: boolean): Promise<string[]> {
```

Comment:
adding the managed policy id fetching here feels a bit weird since the purpose of this method is to get all policy ids. my preference is to do [this]() and use `agentPolicyService.list` (with [kuery arg]() for `filtering is_managed`) [directly in the handler]() so we reduce agent policy specific logic (`is_managed`) inside of uninstall token logic.